prov/rxm/src/rxm_cq.c

```diff
@@ -721,23 +728,59 @@ ssize_t rxm_sar_handle_segment(struct rxm_rx_buf *rx_buf)
 	return rxm_cq_handle_seg_data(rx_buf);
 }
 
+static ssize_t rxm_rndv_send_ack_inject(struct rxm_rx_buf *rx_buf)
+{
+	struct rxm_pkt pkt;
+	struct iovec iov = {
+		.iov_base = &pkt,
+		.iov_len = sizeof(pkt),
+	};
+	struct fi_msg msg = {
+		.msg_iov = &iov,
+		.iov_count = 1,
+		.context = rx_buf,
+	};
+
+	assert(rx_buf->conn);
+
+	pkt.hdr.op		= ofi_op_msg;
+	pkt.hdr.version		= OFI_OP_VERSION;
+	pkt.ctrl_hdr.version	= RXM_CTRL_VERSION;
+	pkt.ctrl_hdr.type	= rxm_ctrl_rndv_ack;
+	pkt.ctrl_hdr.conn_id 	= rx_buf->conn->handle.remote_key;
+	pkt.ctrl_hdr.msg_id 	= rx_buf->pkt.ctrl_hdr.msg_id;
+
+	return fi_sendmsg(rx_buf->conn->msg_ep, &msg, FI_INJECT);
```

Comment:
Can you expand on the commit message to describe the situation where it causes issues?

If fi_inject() eventually fails, it seems like that results in the connection being broken.

Reply:
If the app exits right after sending this ACK with fi_inject, it would result in app reporting success when the inject might in fact have failed or in progress. This is more of a pro-active fix and I haven't seen such an issue yet. In general, I'm looking to replace all protocol messages that use fi_inject with fi_sendmsg + FI_INJECT.